### PR TITLE
Fix coverage reports

### PR DESCRIPTION
**Pull request by michaelkleinhenz**
This fixes the coverage reports..

### CODE REVIEW

File: make/test.mk [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
 ## runs all tests with bundles assets
 test: generate
-	go test -count=1 ./...
+	# test the test package seperately, no coverage report
+	go test -count=1 ./test
+	# test the rest of the code, with coverage report	
+	@-mkdir -p $(COV_DIR)
+	@-rm -f $(COV_DIR)/coverage.txt
+	@-for d in $(GOPKGS) ; do \
+	  go test -coverprofile=profile.out -covermode=atomic -coverpkg=$$GOPKGS $$d ; \
```
Comment (alexeykazakov) [inline make/test.mk]: You don't have to test each package individually when using coverpkg. Just run go test once with that flag.

File: make/test.mk [hunk 2]
```diff
@@ -1,4 +1,18 @@
+COV_DIR = coverage
+GOPKGS = $(shell go list ./... | grep -v /test)
+
 .PHONY: test
 ## runs all tests with bundles assets
 test: generate
-	go test -count=1 ./...
+	# test the test package seperately, no coverage report
+	go test -count=1 ./test
```
Comment (alexeykazakov) [inline make/test.mk]: Also, it's actually fine to include "test" package in the coverage report if it's utility for tests which also covered by tests. Which is the case.
So, let's just run "go test" once for everything. I just thought that "test" package is a utility for tests which doesn't have any tests (to test test utilities) at all.

Comment (michaelkleinhenz) [reply make/test.mk]: At least now I know how Make expands bash commands :-)

Comment (michaelkleinhenz) [reply make/test.mk]: Fixed 4b40afb.

File: make/test.mk (hunk 1 above)
Comment (michaelkleinhenz) [reply make/test.mk]: Fixed 4b40afb.

**Pull request merged**